### PR TITLE
Common update

### PR DESCRIPTION
**Pull request by StKostyantin**
This pull request will update the Client and Server modules. Also, update createReduxStore.

### CODE REVIEW

File: modules/core/common/createReduxStore.ts [hunk 1]
```diff
@@ -8,11 +8,63 @@ export const getStoreReducer = (reducers: any) =>
...
+
+  const { REQUEST, SUCCESS, FAIL } = types;
+
+  next({ type: REQUEST || null, ...rest });
+
+  const handleAPICall = async () => {
+    try {
+      const result = await APICall();
+      const data = result && result.data;
+      if (data.errors) {
```
Comment (evgenijolshanskij) [inline modules/core/common/createReduxStore.ts]: According to this `if` statement the error can be sent via either data in response with status `200` or response with any other statuses. I don't think it makes any sense and we should have just one flow for sending errors.

File: modules/core/common/createReduxStore.ts [hunk 2]
```diff
@@ -8,11 +8,63 @@ export const getStoreReducer = (reducers: any) =>
     ...reducers
   });
 
-const createReduxStore = (reducers: Reducer, initialState: DeepPartial<any>, routerMiddleware?: Middleware): Store => {
+const requestMiddleware: Middleware = _state => next => action => {
+  const { types, APICall, ...rest } = action;
+  if (!types) {
```
Comment (evgenijolshanskij) [inline modules/core/common/createReduxStore.ts]: Not sure I understood the role of the `types` object and why it's optional.

File: modules/core/common/createReduxStore.ts [hunk 3]
```diff
@@ -8,11 +8,63 @@ export const getStoreReducer = (reducers: any) =>
...
+
+const createReduxStore = (
+  reducers: Reducer,
+  initialState: DeepPartial<any>,
+  routerMiddleware?: Middleware,
+  reduxMiddlewares?: Middleware[]
+): Store => {
+  const middleware: () => Middleware[] = () => {
+    const routerMiddlewares = routerMiddleware ? [routerMiddleware] : [];
+    const reduxMiddleware = reduxMiddlewares && reduxMiddlewares.length ? reduxMiddlewares : [];
```
Comment (evgenijolshanskij) [inline modules/core/common/createReduxStore.ts]: Simplify: `const reduxMiddleware = reduxMiddlewares || [];`

Comment (evgenijolshanskij) [reply modules/core/common/createReduxStore.ts]: The array called as a single, while it should be plural.

File: modules/core/common/createReduxStore.ts [hunk 4]
```diff
@@ -8,11 +8,63 @@ export const getStoreReducer = (reducers: any) =>
...
+  };
+
+  return handleAPICall();
+};
+
+const createReduxStore = (
+  reducers: Reducer,
+  initialState: DeepPartial<any>,
+  routerMiddleware?: Middleware,
+  reduxMiddlewares?: Middleware[]
```
Comment (evgenijolshanskij) [inline modules/core/common/createReduxStore.ts]: Isn't it easier and cleaner to use default values?

File: modules/core/client-react-native/App.tsx [hunk 5]
```diff
@@ -24,13 +24,15 @@ export default class Main extends React.Component<MainProps> {
         ? `${protocol}//${url.parse(this.props.exp.manifest.bundleUrl).hostname}:${port}${pathname}`
         : __API_URL__;
 
-    const store = createStore(
+    const store = createReduxStore(
       Object.keys(modules.reducers).length > 0
-        ? combineReducers({
+        ? {
             ...modules.reducers
```
Comment (evgenijolshanskij) [inline modules/core/client-react-native/App.tsx]: Why do you create a duplication for `reducers` object?

File: modules/core/common/createReduxStore.ts (hunk 3 above)
Comment (StKostyantin) [reply modules/core/common/createReduxStore.ts]: Fixed

File: modules/core/common/createReduxStore.ts (hunk 2 above)
Comment (Iwasherd) [reply modules/core/common/createReduxStore.ts]: _types_ is the optional property for asynchronous dispatches.
Synchronous dispatches should have only _type_ property

File: modules/core/common/createReduxStore.ts [hunk 6]
```diff
@@ -8,11 +8,59 @@ export const getStoreReducer = (reducers: any) =>
...
+      const data = result && result.data;
+      next({
+        ...rest,
+        type: SUCCESS || null,
+        payload: data
+      });
+
+      return data;
+    } catch (e) {
+      if (e.response && e.response.data && e.response.data.status === 401) {
```
Comment (evgenijolshanskij) [inline modules/core/common/createReduxStore.ts]: Did you test this code? Are you sure the `status` is attached to `data`?

File: modules/core/common/createReduxStore.ts (hunk 4 above)
Comment (StKostyantin) [reply modules/core/common/createReduxStore.ts]: Fixed

File: modules/core/client-react-native/App.tsx (hunk 5 above)
Comment (Iwasherd) [reply modules/core/client-react-native/App.tsx]: Done

File: modules/core/common/createReduxStore.ts (hunk 6 above)
Comment (Iwasherd) [reply modules/core/common/createReduxStore.ts]: Changed

**Pull request merged**